packages/components/src/components/SquiggleEditor.tsx

```diff
@@ -145,40 +66,36 @@ export interface SquigglePartialProps {
 }
 
 export const SquigglePartial: React.FC<SquigglePartialProps> = ({
-  initialSquiggleString = "",
+  code,
+  defaultCode = "",
   onChange,
+  onCodeChange,
   bindings = defaultBindings,
   environment,
   jsImports = defaultImports,
 }: SquigglePartialProps) => {
-  const [code, setCode] = useState(initialSquiggleString);
-  React.useEffect(
-    () => setCode(initialSquiggleString),
-    [initialSquiggleString]
-  );
+  const [uncontrolledCode, setCode] = useState(defaultCode);
+  let codeProp = code ?? uncontrolledCode;
```

Comment:
This is counterintuitive, I expected `codeProp` to be "code that came from props".

I don't care that much, though, but I do care about consistency - the way I named the vars in `SquigglePlayground` is slightly different and we should do it in the same way in both cases.

We should just write a custom hook, actually, it would make everything much clearer.

```typescript
const [code, setCode] = useMaybeControlledValue({
  value: code,
  defaultValue: defaultCode,
  onChange: onCodeChange,
});
```